src/offline/cable_driver_init.F90

```diff
@@ -0,0 +1,117 @@
+MODULE cable_driver_init_mod
+  !! Module for CABLE offline driver initialisation.
+  USE cable_common_module, ONLY : &
+    filename,                     &
+    calcsoilalbedo,               &
+    redistrb,                     &
+    wiltParam,                    &
+    satuParam,                    &
+    cable_user,                   &
+    gw_params
+  USE cable_IO_vars_module, ONLY : &
+    soilparmnew,                   &
+    output,                        &
+    patchout,                      &
+    check,                         &
+    verbose,                       &
+    leaps,                         &
+    logn,                          &
+    fixedCO2,                      &
+    ncciy,                         &
+    gswpfile,                      &
+    globalMetfile
+  USE casadimension, ONLY : icycle
+  USE casavariable, ONLY: casafile
+  USE cable_namelist_util, ONLY : &
+    get_namelist_file_name,       &
+    CABLE_NAMELIST
+#ifdef __MPI__
+  USE mpi
+  USE cable_mpicommon, ONLY : comm, rank
+#endif
+  IMPLICIT NONE
+  PRIVATE
+
+  LOGICAL, SAVE, PUBLIC :: vegparmnew    = .FALSE. ! using new format input file (BP dec 2007)
+  LOGICAL, SAVE, PUBLIC :: spinup        = .FALSE. ! model spinup to soil state equilibrium?
+  LOGICAL, SAVE, PUBLIC :: spincasa      = .FALSE. ! TRUE: CASA-CNP Will spin mloop times, FALSE: no spin up
+  LOGICAL, SAVE, PUBLIC :: l_casacnp     = .FALSE. ! using CASA-CNP with CABLE
+  LOGICAL, SAVE, PUBLIC :: l_landuse     = .FALSE. ! using CASA-CNP with CABLE
+  LOGICAL, SAVE, PUBLIC :: l_laiFeedbk   = .FALSE. ! using prognostic LAI
+  LOGICAL, SAVE, PUBLIC :: l_vcmaxFeedbk = .FALSE. ! using prognostic Vcmax
+
+  REAL, SAVE, PUBLIC :: delsoilM ! allowed variation in soil moisture for spin up
+  REAL, SAVE, PUBLIC :: delsoilT ! allowed variation in soil temperature for spin up
+  REAL, SAVE, PUBLIC :: delgwM = 1e-4
```

Comment:
This module could be a good place to stop using module variables for passing data between function, which is considered bad practice. Instead, data should be passed as function arguments.

Reply:
FYI I initially considered defining these module variables as function arguments to `cable_driver_init` but I realised the variables were not defined at the top level scope of `cable_mpidrv.F90` which prevents me from passing in these variables as arguments. We could have the master and worker drivers call `cable_driver_init` separately instead of in `cable_mpidrv.F90` so that the variables were in scope but we would need to take out the MPI specific initialisation out of `cable_driver_init` as it would be called after the split between master and worker. Given the other namelist variables are module variables, I figured that would be the easiest workaround.

Reply:
Thanks for the explanation. Let's leave this as it is for now. We can revisit this when all drivers are merged.